Sources/PackageGraph/RepositoryPackageContainerProvider.swift

```diff
@@ -362,7 +362,27 @@ public class RepositoryPackageContainer: BasePackageContainer, CustomStringConve
         // Compute the map of known versions.
         //
         // FIXME: Move this utility to a more stable location.
-        let knownVersions = Git.convertTagsToVersionMap(repository.tags)
+        let knownVersionsWithDuplicates = Git.convertTagsToVersionMap(repository.tags)
+        
+        let knownVersions = knownVersionsWithDuplicates.mapValues { (tags) -> String in
+            switch tags {
+            case (normal: .some, vPrefixed: .none):
+                return tags.normal!
+            case (normal: .none, vPrefixed: .some):
+                return tags.vPrefixed!
+            case (normal: .some, vPrefixed: .some):
+                let normalRevision = try! repository.resolveRevision(tag: tags.normal!)
+                let vPrefixedRevision = try! repository.resolveRevision(tag: tags.vPrefixed!)
+                if normalRevision != vPrefixedRevision {
+                    // Warn if the duplicate tags point to different revisions.
+                    print("Warning: tags \(tags.normal!) and \(tags.vPrefixed!) point to different revisions. Defaulting to \(tags.normal!)")
```

Comment:
We have DiagnosticsEngine to emit warnings but its probably difficult to get that here. I would say just ignore emitting warning for now and leave a FIXME note.

Reply:
Got it!